lib/Helper/DataSetUploadHandler.php

```diff
@@ -90,16 +90,26 @@ protected function handleFormData($file, $index)
             if ($overwrite == 1)
                 $dataSet->deleteData();
 
-            // Load the file
-            ini_set('auto_detect_line_endings', true);
-
             $firstRow = true;
             $i = 0;
 
             $handle = fopen($controller->getConfig()->getSetting('LIBRARY_LOCATION') . 'temp/' . $fileName, 'r');
             while (($data = fgetcsv($handle)) !== FALSE ) {
                 $i++;
 
+                // Filter out rows that are entirely empty
+                $filteredData = array_filter($data, function($row) {
```

Comment:
Are we sure this should be double `array_filter` ?

and not just
```
$filteredData = array_filter($data, function($value) {
    return !empty($value);
});
```

I think the `$data` here is already an array containing single row data

Should this filter also be after the firstRow check? - no need to filter first row if its going to be ignored anyway

Reply:
Your right, I was trying to remember why I did that in the first place but i can't seem to fully remember it. 😅

I think I may have done that to avoid something like `,,,,` or basically lines with no values but has multiple delimiters. And I might have been confused thinking that the `$data` contains all the rows from the CSV when in fact, it only contains a single row per iteration.

Thank you for pointing that out to me.